django/contrib/contenttypes/fields.py

```diff
@@ -359,10 +359,43 @@ def resolve_related_fields(self):
         self.to_fields = [self.model._meta.pk.name]
         return [(self.remote_field.model._meta.get_field(self.object_id_field_name), self.model._meta.pk)]
 
+    def _get_path_info_with_parent(self):
+        """
+        Returns the path with joining through the parent model.
```

Comment:
We return a path that has first a join to the parent model, then to the current model. The idea is that if you have a GFK defined on parent model, then we need to join the parent model first, then the child model.